tests/unit/test_collector.py

```diff
@@ -98,12 +105,14 @@ def test_get_html_response_archive_to_http_scheme_is_html(url):
     assert resp is not None
     assert session.mock_calls == [
         mock.call.head(url, allow_redirects=True),
-        mock.call.head().raise_for_status(),
         mock.call.get(url, headers={
             "Accept": "text/html", "Cache-Control": "max-age=0",
         }),
-        mock.call.get().raise_for_status(),
     ]
+    mock_raise_for_status.assert_has_calls([
+        mock.call(session.head.return_value),
+        mock.call(resp)
+    ])
```

Comment:
I would use `mock_raise_for_status.mock_calls == [...]` instead. IIRC `assert_has_calls()` only checks all given exceptions are raised, but here we also want to ensure *only* those are raised.